packages/react/src/use-table/helpers.tsx

```diff
@@ -138,16 +138,24 @@ export const getTableSelectionMap = (spec: TableSpec) => {
   return selectionMap;
 };
 
-export const getTableRows = (spec: Pick<TableSpec, "fieldMetadataTree" | "targetColumns">, records: GadgetRecord<any>[]) => {
-  return records.map((record, index) => recordToRow(spec, record, index));
+export const getTableRows = (
+  spec: Pick<TableSpec, "fieldMetadataTree" | "targetColumns">,
+  columns: TableColumn[],
+  records: GadgetRecord<any>[]
+) => {
+  return records.map((record, recordIndex) => recordToRow(spec, columns, record, recordIndex));
 };
 
 export const getTableColumns = (spec: Pick<TableSpec, "fieldMetadataTree" | "targetColumns">) => {
   const columns: TableColumn[] = [];
 
-  for (const targetColumn of spec.targetColumns) {
+  for (const [i, targetColumn] of spec.targetColumns.entries()) {
     if (isCustomCellColumn(targetColumn)) {
+      // Fallback to index if crypto.randomUUID is not available. All browsers have crypto.randomUUID, but test envs don't
+      const identifier = crypto.randomUUID?.() ?? i;
```

Comment:
I think it's a bad idea to have the tests and production code working differently -- maybe we could shim with something like https://github.com/ai/nanoid to keep the package size down?

Reply:
I added this to shim the UUIDs because they need to be deterministic as we also need to reference them in the tests. Pretty cool stuff 


```
  let timesUsedCryptoRandomUUID = 0;
  let uuids: string[] = [];

  const shimCryptoRandomUUID = () => {
    Object.defineProperty(globalThis, "crypto", {
      value: {
        randomUUID: () => {
          timesUsedCryptoRandomUUID++;
          const mostRecentCryptoRandomUUID = `000-000-000-000-${timesUsedCryptoRandomUUID}`;
          uuids.push(mostRecentCryptoRandomUUID);
          return mostRecentCryptoRandomUUID;
        },
      },
    });
  };

  beforeEach(() => {
    shimCryptoRandomUUID();
  });
```